### DISCUSSION TITLE
Discussion: Fall Damage Mechanic

### DISCUSSION THREAD

**Discussion by wail** [OpenTournament/OpenTournament#207] [Game Features]
This is a discussion topic related to OT-#80 issue (now closed as we have a discussion topic here).

**Context**
Traditionally, players will take damage from falls if their speed when landing exceeds a certain threshold, scaling up based on the magnitude by which the threshold is exceeded. In certain cases, players may be able to pick up an item in maps that will reduce or eliminate the possibility of fall damage. Typically the possibility of death by fall damage can also be mitigated via healing items, armors, and certain movement techniques.

**Question**
Does damage from long falls contribute to the game's design in any way, or unnecessarily limit map design?

----
https://github.com/OpenTournament/OpenTournament/issues/80#issuecomment-670295310
@septemalteri responds

> This is an element that could possibly be utilized to reward team coordination such as team boosting, in terms of adding a risk/reward element (Knowing the boost will result in fall damage, but if done accurately and skillfully, should land on the "insert item/powerup denial" ultimately breaking the shift of the game to their advantage)
> 
> Or the opposite: You teamboost your teamate for that denial, but you fail and he ends up splattering his organs all over #general.
> 
> Also consider alternate routes for maps: Where a Flag Carrier returning the flag to his base might make a jump/drop that he knows will cause him to lose health even if he's already low, but at the benefit of changing routes and preventing his opponents from otherwise cutting him off. (Risk/Reward)
> 
> This also creates moments where you're left laughing, whether it's because you got lucky and hit that HP pack before landing, or because you landed on the floor with a not-so-graceful "splat".
> 
> There are probably more examples to be made that I think you the reader can recall while playing, where falling damage (and surviving it, or not) arguably made the play more tense/fun.
> 
> I think the dynamics of falling damage should be seen as an opportunity to create these moments, and not as something limiting when it comes to design.

----

https://github.com/OpenTournament/OpenTournament/issues/80#issuecomment-670306501
@DarkArc responds:

> Might be interesting to see some sort of "roll" mechanic, where a player with good timing can negate significant fall damage.

----

https://github.com/OpenTournament/OpenTournament/issues/80#issuecomment-670378515
@Niklot-OT responds:

> 1. I find it absolutely ridiculous that in QuakeWorld or UT one can die by falling off a ledge with low hp.
> 2. Disabling fall damage might feel good on one end, but then it eliminates the risk/reward kind of factor when jumping off certain heights.
> 3. Disabling fall damage disables all kind of punishment for players who want to jump from places like the sniper tower on DM-Phobos.
> 4. Fall damage makes environment traversal a bit more tactical and forces more careful maneuvering.
> 5. Fall damage gained redundancy with double jumping and more extreme movement introduced in later UT titles, since there was significantly more air time in them than in og UT.
> 6. Ridiculous 2hp fall damage deaths could be prevented by a crippled state, so falling off higher ledges would rather be more slowing over a short period of time than damaging and wouldn't be lethal unless in extreme situations (DM-Phobos sniper tower).
> 7. We could expand on the mechanic of falling and fall damage by movement techniques or character tools designed for mitigating fall damage - wall creeping with some sort of claws maybe?

----

https://github.com/OpenTournament/OpenTournament/issues/80#issuecomment-670777533
@Joelllll responds:

> > 1. Disabling fall damage might feel good on one end, but then it eliminates the risk/reward kind of factor when jumping off certain heights.
> 
> If this is desirable how to add risk to falling without damage? Audio for falling players followed by a crunchier footstep sound when landing? This would be good regardless of damage. Player drops currently equipped weapon forwards some amount based on fall height, etc.
> 
> The "risk" of jumping off something that causes 30hp damage is not really something (imo) that players will be constantly weighing up and only care about it when they are low on health - otherwise they will just eat the damage and then heal up. This is particularly true in ut games with higher player counts where armor is relatively rare (3-5 times per minute) but health packs are much more common due to 20s spawn.
> 
> > 1. Disabling fall damage disables all kind of punishment for players who want to jump from places like the sniper tower on DM-Phobos.
> 
> At the same time it opens up being able to jump from this location which can be super cool as well. Possibly cooler/more interesting than "pay the 50hp to jump down" tax. Obv phobo tower is death..
> 
> Also air time in afps is typically a state where players are more vulerable. Falling allows easier tracking (as do features like bounce pads and lift jumps) so falling could be said to be a disadvantage due to this. Thinking back to point 2 and risk/reward - highground is beneficial, it takes a special something-something to make a player cede highground. Perhaps it is ok not having fall damage to make giving up high ground more appealing in more situations? Not directly punishing the fall as it is already punished by (probable) loss of good posistion.
> 
> > 1. Fall damage makes environment traversal a bit more tactical and forces more careful maneuvering.
> 
> This is not that common really. On most maps this equates to taking small amount of chip damage while moving around and jumping off "normal" sized geometry..
> 
>     1. Ridiculous 2hp fall damage deaths could be prevented by a crippled state, so falling off higher ledges would rather be more slowing over a short period of time than damaging and wouldn't be lethal unless in extreme situations (DM-Phobos sniper tower).
> 
> 
> A cripple effect on landing could be interesting - but also annoying af if it triggers frequently. Perhaps no fall damage for what you would traditionally take damage from and then.. instant death above a certain (reasonable) height?
> 
> If a cripple state were used frequently for lower falls that one might find on normal maps it would potentially be significantly worse than taking the (generally) low damage as a crippled player becomes an easy target. 
> Another thing to keep in mind is slide, which has been requested often. Being able to fall to gain velocity which can then be converted into slide is nice.

----

https://github.com/OpenTournament/OpenTournament/issues/80#issuecomment-679224361
@wail responds:
 
> I'd been holding off on chiming in on this since I didn't want to necessarily influence others, but it's been awhile so here goes.
> 
> **Cons**
> 
>     * Dying from falling damage is often a feelsbad moment
> 
>     * Dying from falling damage doesn't enhance the feel of hyper-badass athletes with advanced sci-fi equipment
> 
>     * Falling damage might be seen as enforcing arbitrary limitations on map design
> 
> 
> **Pros**
> 
>     * Intuitive. When you fall a long distance and take damage no one thinks this is unusual
> 
>     * Scalable effect. There's a difference between falling and taking 5 damage and taking 100 damage.
> 
>     * Some map designs are essentially impossible without falling damage (e.g. CTF-Face completely changes)
> 
>     * Pickups such as Jump Boots (or, newly-proposed 'Featherfall Boots') can mitigate falling damage.
> 
>     * There are multiple strategies available to allow level designers to design maps with extensive use of verticality without having players constantly be under threat of falling damage. (e.g. Water volumes, lots of Featherfall Boot pickups, or using a level blueprint to automatically assign all players a Featherfall Boot item on spawn.)
> 
> 
> Two other important considerations:
> 
>     * Falling damage is already implemented
> 
>     * Falling damage is an expected convention
> 
> 
> Alternate proposals for fall damage mitigation and states like 'cripple' are really beyond the scope of the discussion here because that's proposing a new feature and one that has a lot of questions & vagueness that would need to be clearly answered before we could judge its impact on the game.

**Comment by MonsOlympus** [comment]
Some counter points
- Dying from falling damage is all on you, any similar suicide will feel bad because of your failure.
- Extremely long falls could lead to exploits, if people can combine rocket jumps and other movements like the grapple this could lead to situations no map could realistically account for. (Xonotic Instagib+hooks)
- Designing levels to account for falling damage can be done by simply placing some small ledges for a player to skip down safely as falling damage is velocity based. (On certain slopes some sort of skid/wall run mechanic could be considered.)

In terms of "hyper-badass athletes with advanced sci-fi equipment"
- It makes sense to focus on the ability of said athletes, if anyone could do it then why are these combatants so hyper-badass?
- Same goes for the equipment, if anyone can just roll said gadgets and be amazeballs then whats the draw for the skillful competitive characters?

My suggestion is to focus on making mitigating falling damage a skill, giving players the tools to avoid "feelsbadman" moments and leave it up to them. There are always gameplay modifiers which could be useful like lowgrav mode, some sort of gravbelt (or featherweight boots) but there could be a tradeoff, if that tradeoff is an option given to the player to apply their own skill youre getting a good middle ground between accessibility, skillful play and map support.
